sphinx/ext/autosummary/generate.py

```diff
@@ -151,11 +159,32 @@ def generate_autosummary_docs(sources, output_dir=None, suffix='.rst',
             warn('[autosummary] failed to import %r: %s' % (name, e))
             continue
 
+        # https://docs.python.org/3/reference/import.html#packages
+        # ... any module that contains a __path__ attribute is
+        # considered a package ...
+        ispackage = hasattr(obj, '__path__')
+        if ispackage:
+            depth = len(name.split('.')) - 1
+            recurse_package = depth < recursion_limit or recursion_limit < 0
```

Comment:
It seems `recursion_limit` is compared with the depth of package from root. I think it is not called "recursion".

Reply:
Yes I should specify in the docs that this is recursion depth from root. At first I was thinking of limiting the `generate_autosummary_docs` call depth but I think this gives confusing results, depending on the situation. Let me know if you have an issue with the concept or just the name of the parameter.

Reply:
Called it `autosummary_depth_limit`. Docs mention it is the depth starting from root.